rfc/rfc-73/rfc-73.md

```diff
@@ -0,0 +1,416 @@
+
+**Analysis with MVCC**: This is a **potential problem** even with snapshot isolation. Let's say two transactions, T1 and
+T2, start at the same time and read the same file in one of the tables. Both decide to modify the file based on what
+they've read. Since they are operating on snapshots, neither transaction sees the other's changes. When they try to
+commit, they might both try to create a new version of the file, leading to a write-write conflict. The system, noticing
+the conflict, would typically abort one of the transactions or apply some conflict resolution strategy.
+
+#### Conflict Resolution
+
+Conflict detection will happen on the basis of set of partitions and fileIDs mutated by the transaction. Let's say there
+are two conflicting transaction T1 and T2. Both of them, when they start, fetch latest versions of all tables and
+register themselves with start timestamp in the database timeline, which will be greater than all versions of tables
+involved in T1 and T2. To handle conflict before committing, we have the following options:
+
+1. **First committer/Younger transaction wins** - Essentially, no resolution required with end timestamp based ordering.
+   If a transaction tries to modify data that has been modified by a younger transaction (with a later timestamp), the
+   older transaction is rolled back to avoid the conflict. This ensures that transactions are serialized based on their
+   timestamps.
+    1. Pros: Transactions are serialized in a consistent order. Works well in environments with low contention.
+    2. Cons: Have to maintain end timestamp for each action. Potentially high abort rates in high contention
+       environments.
+
+2. **OCC (with lock provider)** - With start and end timestamp, we have the option of "wait-die" and "
+   wound-wait" ([CMU notes on 2PL section 3](https://15445.courses.cs.cmu.edu/fall2022/notes/16-twophaselocking.pdf))
+   strategies. In the Wait-Die strategy, if an older transaction requests a lock held by a younger one, the older
+   transaction is forced to wait or is aborted ("die") after some time. In the Wound-Wait scheme, the younger
+   transaction is aborted ("wounded") to allow the older transaction to proceed.
+    1. Pros: Provides a mechanism to prioritize older transactions.
+    2. Cons: Complexity in implementation and potentially high waiting time if there are frequent short-running
+       transactions.
+
+3. [Compensating transaction](https://learn.microsoft.com/en-us/azure/architecture/patterns/compensating-transaction) -
+   Instead of aborting a transaction when a conflict is detected, another transaction is executed based on the latest
+   snapshot to compensate for the effects of the conflicting transaction.
+    1. Pros: Avoids the need to abort and retry. Could be useful for long-running transactions.
+    2. Cons: Complexity in designing and ensuring the correctness of compensating transactions.
+
+## Implementation
+
+### **SQL Writes**
```

Comment:
lets make sure we model this after an ANSI compliant database or datawarehouse?

Reply:
Got it. The BEGIN..END is very similar to what we have in MySQL - https://dev.mysql.com/doc/refman/8.0/en/begin-end.html